blob/azureblob/azureblob.go

```diff
@@ -318,8 +318,34 @@ var errUnimplemented = errors.New("not implemented")
 
 // Copy implements driver.Copy.
 func (b *bucket) Copy(ctx context.Context, dstKey, srcKey string, opts *driver.CopyOptions) error {
-	// TODO(rvangent): Implement this and delete errUnimplemented.
-	return errUnimplemented
+	dstKey = escapeKey(dstKey, false)
+	dstBlobURL := b.containerURL.NewBlobURL(dstKey)
+	srcKey = escapeKey(srcKey, false)
+	srcURL := b.containerURL.NewBlobURL(srcKey).URL()
+	resp, err := dstBlobURL.StartCopyFromURL(ctx, srcURL, nil /* metadata */, azblob.ModifiedAccessConditions{}, azblob.BlobAccessConditions{})
+	if err != nil {
+		return err
+	}
+	copyStatus := resp.CopyStatus()
+	nErrors := 0
```

Comment:
This can be defined inside the for loop.

Reply:
It can't be inside the for loop, as it may be incremented inside the for loop. I guess it could be declared as part of the `for` itself, e.g. `for nErrors := 0; copyStatus == copyStatusPending`, but I find that less readable.

Reply:
Oops, didn't know what I was reading...